Jenkinsfile

```diff
@@ -0,0 +1,72 @@
+pipeline {
+    agent {
+        dockerfile {
+            filename 'Dockerfile.build'
+        }
+    }
+
+    stages {
```

Comment:
I haven't quite thought about how to approach this yet, however, we can share parts of Jenkinsfiles between projects (like so: https://github.com/digirati-co-uk/dlcs-environment-pipeline/tree/master/vars). I'd like to do that so we can reduce the duplication and maintenance burden of having [relatively] large Jenkinsfiles.

E.g., we could have something like:
```
genericCodeAnalysis(['javac', 'spotbugs', 'checkstyle'])
```
And we can start using defaults where appropriate.

Reply:
I don't have permissions to view that, but if it's anything like this I'd be down for that:

https://jenkins.io/blog/2017/10/02/pipeline-templates-with-shared-libraries/

Reply:
Ah, yep, that's it. You should be able to access that repo now. They work quite well since you can write automated tests around the pipelines themselves if the steps get large.